### PR TITLE
SMAR-17: js ajax error handling

### PR DESCRIPTION
**Pull request by mant02**
Added error handling for ajax.
Improved parts of logic and added extra validation.

Tested out by replicating errors or exiting with a http 401 on the back end.

### CODE REVIEW

File: smartsupp.php [hunk 1]
```diff
@@ -347,6 +347,12 @@ public function hookDisplayBackOfficeHeader()
         $js = '';
 
         if (strcmp(Tools::getValue('configure'), $this->name) === 0) {
+            \Media::addJsDef([
```
Comment (mandan2) [inline smartsupp.php:356]: What about the other hook where smartsupp.js file is appended? not sure when it loads

File: smartsupp.php [hunk 2]
```diff
@@ -347,6 +347,12 @@ public function hookDisplayBackOfficeHeader()
...
+                'smartsupp' => [
+                    'genericAjaxErrorMessage' => $this->l('Unknown error occurred. Please contact support.'),
```
Comment (mandan2) [inline smartsupp.php]: Try again or contact support

File: views/js/smartsupp.js [hunk 3]
```diff
@@ -106,36 +124,53 @@ jQuery(document).ready( function($) {
...
+                    $("#smartsupp_create_account .alerts").show();
+
+                    if (response.message) {
+                        $("#smartsupp_create_account .alerts .alert").html(response.message);
                     }
+
+                    return;
+                }
+
+                $("#smartsupp_create_account .alerts .alert").html(smartsupp.genericAjaxErrorMessage);
```
Comment (mandan2) [inline views/js/smartsupp.js]: so on success you show error message?

File: views/js/smartsupp.js [hunk 4]
```diff
@@ -106,36 +124,53 @@ jQuery(document).ready( function($) {
             },
             dataType: 'json',
             headers: { "cache-control": "no-cache" },
-            success: function(data) {
-                $("input#smartsupp_key").val(data.key);
```
Comment (mandan2) [inline views/js/smartsupp.js:110]: I see previously these values were used, not needed anymore?

File: views/js/smartsupp.js [hunk 5]
```diff
@@ -106,36 +124,53 @@ jQuery(document).ready( function($) {
...
+            url: ajax_controller_url,
+            async: false,
+            type: 'POST',
+            data: {
+                action: 'deactivate'
+            },
+            dataType: 'json',
+            headers: { "cache-control": "no-cache" },
+            success: function (response) {
+                $("input#smartsupp_key").val(response.key);
```
Comment (mandan2) [inline views/js/smartsupp.js:166]: here we won't have any errors in response body? If they exist, add the same alert block. Ideally, could just adapt how errors are showed above and reuse same blocks. You didn't post any images so not sure how these validations looks like

Comment (mant02) [reply views/js/smartsupp.js:166]: this ajax only removes values from the configuration tab. So no errors or messages are parsed as a response. No api calls either.

File: smartsupp.php (hunk 1 above)
Comment (mant02) [reply smartsupp.php:356]: used for different PS versions. backOfficeHeader (1.6.x), displayBackOfficeHeader > 1.6

**Review by mandan2 — APPROVED** [review]

**Pull request merged**